src/main/java/com/rapid7/client/dcerpc/objects/RPCConformantIntegerArray.java

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright 2017, Rapid7, Inc.
+ *
+ * License: BSD-3-clause
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ * * Redistributions of source code must retain the above copyright notice,
+ * this list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright
+ * notice, this list of conditions and the following disclaimer in the
+ * documentation and/or other materials provided with the distribution.
+ *
+ * * Neither the name of the copyright holder nor the names of its contributors
+ * may be used to endorse or promote products derived from this software
+ * without specific prior written permission.
+ */
+package com.rapid7.client.dcerpc.objects;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import com.rapid7.client.dcerpc.io.PacketInput;
+import com.rapid7.client.dcerpc.io.PacketOutput;
+
+public class RPCConformantIntegerArray extends RPCConformantArray<Integer> {
+
+    @Override
+    public void unmarshalEntity(PacketInput in) throws IOException {
+        if (getMaxCount() >= 0) {
```

Comment:
One thing I've noticed is that you can't always guarantee that this is the size of the array, which is why microsoft made the separate struct with the length as a field. So far this seems to be ok, but seems like you'd want the constructor to take a size instead

Reply:
@tosmun-r7 Then how do we know when we stop reading the next value from the array?

Reply:
The constructor is taking the size from getMaxCount(), or is it a different size that you are referring to?

Reply:
I was imagining the constructor of the rpc conformant array take the size

Reply:
I thought we were using setters. The RPCConformant array would read the size from the packet input.